### ISSUE TITLE
Padding of the Conditional Position Encoding (CPE)

### ISSUE THREAD

**Issue by OuYaozhong** [Pointcept/Pointcept#350]
Hi, the authors of Pointcept,

Recently, I try to understand the whole project of the Pointcept.

When I read the code about the Point Transformer V3, you have adopt the conditional position encoding (CPE).

I see that the `cpe` did not use any padding. You can refer to your code about the [`Block` class in Point Transformer V3](https://github.com/Pointcept/Pointcept/blob/c4363c86e1e77325645d44bc8052b338031f1b30/pointcept/models/point_transformer_v3/point_transformer_v3m1_base.py#L277-L287).

However, in the original paper of CPE (i.e., [Conditional Positional Encodings for Vision Transformers, ICLR'23, Xiangxiang Chu, and etc.](https://arxiv.org/abs/2102.10882)), Section 3.2, the paragraph name "Positional Encoding Generator", it tells that

"
PEG can be efﬁciently implemented with a 2-D convolution with kernel k (k ≥ 3) and k−1 zero paddings. _**Note that the 2 zero paddings here are important to make the model be aware of the absolute positions**_,
"

In short, the original paper said the padding is important for model to sense the absolute position.

So why the `Block.cpe` did not use the padding ?

Or you think that the absolute position information can be provided by the `coord` or `xyz`, so the padding can be ignored ?

Hope for your reply.

**Comment by Gofinge** [comment]
Hi, normal convolution for dense images needs padding to prevent the drop of some pixels during convolution. However, for Submanifold Sparse Convolution we don't need padding.
